rust/src/transformations/make_stable_expr/expr_boolean_function/mod.rs

```diff
@@ -0,0 +1,94 @@
+use polars::prelude::*;
+use polars_plan::dsl::{BooleanFunction, Expr, FunctionExpr};
+use polars_plan::prelude::{ApplyOptions, FunctionOptions};
+
+use crate::core::{ExprFunction, Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{AtomDomain, ExprDomain, OuterMetric};
+use crate::error::*;
+use crate::transformations::DatasetMetric;
+
+use super::StableExpr;
+
+#[cfg(test)]
+mod test;
+
+/// Make a Transformation that returns a boolean function expression
+///
+/// # Arguments
+/// * `input_domain` - Expr domain
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared
+/// * `expr` - The boolean function expression
+pub fn make_expr_boolean_function<M: OuterMetric>(
+    input_domain: ExprDomain,
+    input_metric: M,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, M, M>>
+where
+    M::InnerMetric: DatasetMetric,
+    M::Distance: Clone,
+    (ExprDomain, M): MetricSpace,
+    Expr: StableExpr<M, M>,
+{
+    let Expr::Function {
+        input,
+        function: FunctionExpr::Boolean(bool_function),
+        ..
+    } = expr
+    else {
+        return fallible!(MakeTransformation, "expected boolean function expression");
+    };
+
+    let Ok([input]) = <&[_; 1]>::try_from(input.as_slice()) else {
+        return fallible!(
+            MakeTransformation,
+            "{} must have one argument",
+            bool_function
+        );
+    };
+
+    let t_prior = input
+        .clone()
+        .make_stable(input_domain.clone(), input_metric.clone())?;
+    let (middle_domain, middle_metric) = t_prior.output_space();
+
+    let mut output_domain = middle_domain.clone();
+    let active_series = output_domain.active_series_mut()?;
+
+    use BooleanFunction::*;
+
+    match bool_function {
+        IsNull | IsNotNull => active_series.nullable = false,
+        IsFinite | IsInfinite | IsNan | IsNotNan | Not => (),
+        _ => {
+            return fallible!(
+                MakeTransformation,
+                "{:?} is not a recognized row-by-row boolean function",
```

Comment:
In another PR, I think the messaging emphasized that a failed match just wasn't something we'd implemented yet. This makes it sound more like a user error.